HW1/P3/P3.py

```diff
@@ -0,0 +1,50 @@
+
+
+import math as ma
+import findspark
+import os
+findspark.init() 
+import pyspark
+sc = pyspark.SparkContext()
+import numpy as np
+
+import string
+import itertools
+path = '/Users/patrickkuiper/Desktop/Academic/Harvard Courses/CS205/HW/HW1/EOWL-v1.1.2/CSV Format/'
+letters = list(string.ascii_uppercase)
+file_end = ' Words.csv'
+word_matrix = []
+for i in range(len(letters)):
+    word_matrix.append(np.genfromtxt(path+letters[i]+file_end, delimiter=',', dtype = str))
+
+
+word_list = itertools.chain(*word_matrix)
+flat_word_list = list(word_list)
+
+
+word_rdd = sc.parallelize(flat_word_list)
+word_kv = word_rdd.map(lambda x: (''.join(sorted(list(x))), x)).repartition(100)
+word_val = word_rdd.map(lambda x: (''.join(sorted(list(x))), 1)).repartition(100).reduceByKey(lambda a, b: a + b)

```

Comment:
I see why you made this, but this probably could have been accomplished without creating a separate function. Note that you are doing the expensive ''.join(sorted(list(x)) operation both on this line and the one above.